### PR TITLE
Simple wrapper script + hierarchical Makefiles.

### PR DESCRIPTION
**Pull request by NGenetzky**
Very simple wrapper than only determines which executable to call. Also moved the Makefile organization to a hierarchical organization

### CODE REVIEW

File: scripts/cfmask.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        '''
+        satellite_code = xml_filename[0:3]
+        l8_prefixes = ['LC8', 'LO8']
+        if satellite_code in l8_prefixes:
+            return True
+        else:
+            return False
+
+
+class EXECUTE_ERROR(Exception):
```
Comment (dhill-usgs) [inline scripts/cfmask.py]: classes are not all caps

File: scripts/Makefile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#
+# Simple makefile for installing the scripts.
+#------------------------------------------------------------------------------
+
+all:
+
+install:
+	install -d $(PREFIX)/bin
+	install -m 775 cfmask.py $(PREFIX)/bin
+clean:
```
Comment (rdilley-usgs) [inline scripts/Makefile]: Add a space before clean:

File: scripts/cfmask.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    cmd.extend(sys.argv[1:])  # Pass all arguments through
+    cmd_string = ' '.join(cmd)
+    try:
+        logger.info('>>'+cmd_string)
+        output = execute_cmd(cmd_string)
+
+        if len(output) > 0:
+            logger.info("\n{0}".format(output))
+    except EXECUTE_ERROR:
+        logger.exception(('Error running {0}.'
```
Comment (rdilley-usgs) [inline scripts/cfmask.py]: Don't need the extra parens wrapping the string for the format command.  (('Error   ....  ).format

File: scripts/cfmask.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        (status, output) = commands.getstatusoutput(cmd_string)
+
+        if status < 0:
+            message = "Application terminated by signal [%s]" % cmd_string
+            if len(output) > 0:
+                message = ' Stdout/Stderr is: '.join([message, output])
+            raise EXECUTE_ERROR(message)
+
+        if status != 0:
+            message = "Application failed to execute [%s]" % cmd_string
```
Comment (rdilley-usgs) [inline scripts/cfmask.py]: Switch this to use ''.format, as well as single quotes.

File: scripts/cfmask.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        Raises:
+            EXECUTE_ERROR (Stdout/Stderr)
+
+        Returns:
+            output:The stdout and/or stderr from the executed command.
+        '''
+        (status, output) = commands.getstatusoutput(cmd_string)
+
+        if status < 0:
+            message = "Application terminated by signal [%s]" % cmd_string
```
Comment (rdilley-usgs) [inline scripts/cfmask.py]: Switch this to use ''.format, as well as single quotes.

File: scripts/cfmask.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            raise EXECUTE_ERROR(message)
+
+        if status != 0:
+            message = "Application failed to execute [%s]" % cmd_string
+            if len(output) > 0:
+                message = ' Stdout/Stderr is: '.join([message, output])
+            raise EXECUTE_ERROR(message)
+
+        if os.WEXITSTATUS(status) != 0:
+            message = ("Application [%s] returned error code [%d]"
```
Comment (rdilley-usgs) [inline scripts/cfmask.py]: Switch this to use ''.format, as well as single quotes.

File: scripts/cfmask.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+    cmd = [get_executable(isLandsat8)]
+    cmd.extend(sys.argv[1:])  # Pass all arguments through
+    cmd_string = ' '.join(cmd)
+    try:
+        logger.info('>>'+cmd_string)
+        output = execute_cmd(cmd_string)
+
+        if len(output) > 0:
+            logger.info("\n{0}".format(output))
```
Comment (rdilley-usgs) [inline scripts/cfmask.py]: Use single quotes

File: scripts/cfmask.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    logger = get_logger()
+
+    xml_filename = parse_only_xml()
+    isLandsat8 = is_landsat8(xml_filename)
+
+    cmd = [get_executable(isLandsat8)]
+    cmd.extend(sys.argv[1:])  # Pass all arguments through
+    cmd_string = ' '.join(cmd)
+    try:
+        logger.info('>>'+cmd_string)
```
Comment (rdilley-usgs) [inline scripts/cfmask.py]: Use ''.format here instead of +

File: scripts/cfmask.py (hunk 1 above)
Comment (NGenetzky) [inline scripts/cfmask.py]: It was originally a constant. Fixed.
```class ExecuteError(Exception):```

File: scripts/Makefile (hunk 2 above)
Comment (NGenetzky) [inline scripts/Makefile]: Fixed.

File: scripts/cfmask.py (hunk 3 above)
Comment (NGenetzky) [inline scripts/cfmask.py]: Fixed.
```
logger.exception('Error running {0}.'
                         'Processing will terminate.'
                         .format(os.path.basename(__file__)))
```

**Comment by NGenetzky** [comment]
Fixed all single quotes. Also changed all %s string modification to str.format().
```
if status < 0:
            message = ('Application terminated by signal [{0}]'
                       .format(cmd_string))
            if len(output) > 0:
                message = ' Stdout/Stderr is: '.join([message, output])
            raise ExecuteError(message)

        if status != 0:
            message = 'Application failed to execute [{0}]'.format(cmd_string)
            if len(output) > 0:
                message = ' Stdout/Stderr is: '.join([message, output])
            raise ExecuteError(message)

        if os.WEXITSTATUS(status) != 0:
            message = ('Application [{0}] returned error code [{1}]'
                       .format(cmd_string, os.WEXITSTATUS(status)))
            if len(output) > 0:
                message = ' Stdout/Stderr is: '.join([message, output])
            raise ExecuteError(message)
```

**Pull request closed without merging**